From 0eaf5e3d049054c12775a94fa4ad783d927ec084 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 28 Sep 2017 20:57:38 +0800
Subject: [PATCH 3/8] sunxi: update MMIO addresses for R40 display pipeline

The MMIO addresses of the HDMI controller on Allwinner R40 is the same
with other SoCs with obfuscated DesignWare HDMI, but the TCONs are
complex and the addresses are different from the SoCs before it.

Update the MMIO addresses of HDMI and TCONs for R40.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 arch/arm/include/asm/arch-sunxi/cpu_sun4i.h | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/arch/arm/include/asm/arch-sunxi/cpu_sun4i.h b/arch/arm/include/asm/arch-sunxi/cpu_sun4i.h
index 2419062d45..765eeb0374 100644
--- a/arch/arm/include/asm/arch-sunxi/cpu_sun4i.h
+++ b/arch/arm/include/asm/arch-sunxi/cpu_sun4i.h
@@ -38,8 +38,10 @@
 #define SUNXI_TVE0_BASE			0x01c0a000
 #endif
 #define SUNXI_EMAC_BASE			0x01c0b000
+#ifndef CONFIG_MACH_SUN8I_R40
 #define SUNXI_LCD0_BASE			0x01c0C000
 #define SUNXI_LCD1_BASE			0x01c0d000
+#endif
 #define SUNXI_VE_BASE			0x01c0e000
 #define SUNXI_MMC0_BASE			0x01c0f000
 #define SUNXI_MMC1_BASE			0x01c10000
@@ -50,7 +52,8 @@
 #define SUNXI_USB1_BASE			0x01c14000
 #endif
 #define SUNXI_SS_BASE			0x01c15000
-#if !defined(CONFIG_MACH_SUNXI_H3_H5) && !defined(CONFIG_MACH_SUN50I)
+#if !defined(CONFIG_MACH_SUNXI_H3_H5) && !defined(CONFIG_MACH_SUN50I) && \
+	!defined(CONFIG_MACH_SUN8I_R40)
 #define SUNXI_HDMI_BASE			0x01c16000
 #endif
 #define SUNXI_SPI2_BASE			0x01c17000
@@ -158,6 +161,12 @@ defined(CONFIG_MACH_SUN50I)
 #define SUNXI_DRAM_PHY0_BASE		0x01c65000
 #define SUNXI_DRAM_PHY1_BASE		0x01c66000
 
+#define SUNXI_TCON_TOP_BASE		0x01c70000
+#define SUNXI_TCON_LCD0_BASE		0x01c71000
+#define SUNXI_TCON_LCD1_BASE		0x01c72000
+#define SUNXI_TCON_TV0_BASE		0x01c73000
+#define SUNXI_TCON_TV1_BASE		0x01c74000
+
 #define SUNXI_GIC400_BASE		0x01c80000
 
 /* module sram */
@@ -178,7 +187,8 @@ defined(CONFIG_MACH_SUN50I)
 #define SUNXI_MP_BASE			0x01e80000
 #define SUNXI_AVG_BASE			0x01ea0000
 
-#if defined(CONFIG_MACH_SUNXI_H3_H5) || defined(CONFIG_MACH_SUN50I)
+#if defined(CONFIG_MACH_SUNXI_H3_H5) || defined(CONFIG_MACH_SUN50I) \
+	|| defined(CONFIG_MACH_SUN8I_R40)
 #define SUNXI_HDMI_BASE			0x01ee0000
 #endif
 
-- 
2.13.6

